LuaOpenGL: Add POINT_SMOOTH const and query.

saurtron opened the pull request:
### Work done

- Add POINT_SMOOTH const and query capability.

### Remarks

- In some situations users might need to set/query/restore the POINT_SMOOTH to avoid points rendering as squares.
- Unsure if this is wanted since I guess modern opengl doesn't recommend drawing points like that, or maybe it's even deprecated, but couldn't find where it says to be deprecated so proposing this since it can be needed to easy fix widgets rendering points as squares on some gfx cards (like ping wheel).
  - This isn't strictly needed, but specially the query capability is important so state can be restored and not break anything for others. I'm guessing it must be disabled generally, but can't know for sure without being able to query.
- Didn't add a setter since it can be awkwardly done through `gl.UnsafeState`, but could add it too if preferred.

sprunk commented:
Set and get should be paired unless there's precedent for not doing that.

lhog commented:
Point smoothing is generally considered deprecated, because it's a fixed function pipeline aka FFP. Game devs are expected to use shaders to emulate it https://stackoverflow.com/questions/27098315/render-large-circular-points-in-modern-opengl

saurtron commented:
> Point smoothing is generally considered deprecated, because it's a fixed function pipeline aka FFP. Game devs are expected to use shaders to emulate it https://stackoverflow.com/questions/27098315/render-large-circular-points-in-modern-opengl

Thought so, that's why I didn't add the setter... anyways it can actually be set (through `gl.UnsafeState`) and being used in the wild, so I think it'd be better to allow querying the value to avoid problems... Otherwise maybe we should add smth to facilitate modern drawing of points since tbh drawing points is such a hassle when you just want to draw one circle with modern gl :P.

I understand applications are not supposed to be drawing like that, but since migrating that one is so tricky and ppl might not always have the mana for that, thought having this can help in fixing things when circles start rendering as squares.

Anyways I understand if this doesn't get merged because of that.

> Set and get should be paired unless there's precedent for not doing that.

Yeah, I know, just didn't want to facilitate using it too much tbh. Anyways adding it for now.

lhog commented:
Do you have a specific use case in mind?

saurtron commented:
> If `GL_POINT_SMOOTH` is deprecated, and setting is already available via UnsafeState, then not adding a dedicated Set sounds good. In that case maybe it would be good to add some sort of deprecated tag to the Get so that gamedevs are aware it's just for maintaining legacy widgets rather than for any new functionality.

ok, removed the setter and added a deprecation stance to the getter.

lhog commented:
> Yeah it's for some user widgets like the bar ping wheel (was coded by someone else). I had to do a quick fix for some users, but am afraid to apply to everyone since I can't restore the previous value and don't know what it is since I can't query it.

So they used something like `gl.Enable(0x0B10)` in their code?
